chore: [IUM-3387] pin dependency versions

joel-hamilton opened the pull request:
## ✏️ Changes
  
This PR pins some dependency versions in preparation for upgrading to node22. 

It also adds deploy tooling. I test deployed this code to:
- https://cdn.auth0.com/extensions/auth0-ldap-conector-health-monitor/develop/auth0-ldap-conector-health-monitor-2.2.js
- https://cdn.auth0.com/extensions/auth0-ldap-conector-health-monitor/develop/auth0-ldap-conector-health-monitor-2.2.0.js
  
## 📷 Screenshots
 
  
## 🔗 References
  
[IUM-3387](https://auth0team.atlassian.net/browse/IUM-3387)
  
## 🎯 Testing
  
Versions are verified with [canirequire](https://auth0-extensions.github.io/canirequire) to work with both node12 (current runtime) and node22. I've also tested these pinned versions manually with node12 and node22.

### Node 12 with pinned deps
<img width="953" alt="image" src="https://github.com/user-attachments/assets/90b5ab6d-11d7-42ab-a2e0-82d769e0da42" />

### Node 22 with pinned deps
<img width="949" alt="image" src="https://github.com/user-attachments/assets/574e9913-916c-4fab-b959-317f2f1d85c6" />


✅ This change has been tested in a Webtask
 
🚫 This change has unit test coverage
  
🚫 This change has integration test coverage
  
🚫 This change has been tested for performance
  
## 🚀 Deployment
    
✅ This can be deployed any time
 
  
## 🎡 Rollout
  
In order to verify that the deployment was successful we will …
  
## 🔥 Rollback
    
We will rollback if …
  
### 📄 Procedure
  
Downgrade to previous version of (`ldap-connector-health-monitor@2.1.2`) in auth0-extensions. This involves changing the `extensions.json` file and then getting [#dx-eco-extensibility-dev](https://okta.enterprise.slack.com/archives/C01TA7U9X3M) to deploy it.
 
## 🖥 Appliance
  
**Note to reviewers:** ensure that this change is compatible with the Appliance.

joel-hamilton committed:
chore: pin dependency versions

gausnes commented on webtask.json:
```diff
@@ -1,7 +1,7 @@
 {
   "title": "Auth0 AD/LDAP Connector Health Monitor",
   "name": "auth0-ldap-conector-health-monitor",
-  "version": "2.1.2",
```
Looking to force rollout these changes?

joel-hamilton replied:
Nope! There will be a runtime upgrade to this version, so it should definitely be opt-in. 

Changed this to be a minor version bump now, customers should see something like below.

![image](https://github.com/user-attachments/assets/bd4d6268-0278-4faf-ab49-0594abb6eb6e)

bernardwolff replied:
Don't forget to create a tag in the repo after this PR is merged.

bernardwolff replied:
https://github.com/auth0-extensions/auth0-ldap-connector-health-monitor/tags

joel-hamilton committed:
chore: bump minor version

We need to bump the minor version, not the patch version. That way the new extension runtime is opt-in for customers.
Ref: https://oktawiki.atlassian.net/wiki/spaces/Bacca/pages/3119582012/Running+extensions+in+a+specified+runtime#What-do-I-need-to-do-to-have-my-extension-run-in-a-specific-runtime%3F

joel-hamilton committed:
chore: add new deploy action

The previous deploy tooling [auth0-extensions-deploy](https://github.com/atko-cic/auth0-extensions-deploy)
hasn't been working and may be officially dead:
- https://auth0.slack.com/archives/C06KM2779U5/p1736268213432309
- https://auth0.slack.com/archives/C06KM2779U5/p1745607523936019

bernardwolff commented on tools/cdn.sh:
```diff
@@ -0,0 +1,42 @@
+# Use shell parameter expansion to get the major.minor version
+# ${variable%pattern} removes the shortest matching suffix pattern.
+# '.*' matches the last dot and anything following it.
+MAJORMINOR_VERSION_ONLY=${FULL_VERSION%.*}
```
nice!

bernardwolff commented on tools/cdn.sh:
```diff
@@ -0,0 +1,42 @@
+# Extract major.minor version
+# Use shell parameter expansion to get the major.minor version
+# ${variable%pattern} removes the shortest matching suffix pattern.
+# '.*' matches the last dot and anything following it.
```
nice!

bernardwolff commented on tools/cdn.sh:
```diff
@@ -0,0 +1,42 @@
+  if [ ! -z "$BUNDLE_EXISTS" ] && [ "$1" == "$CURRENT_VERSION" ]; then
+    echo "There is already a $EXTENSION_NAME-$1.js in the cdn. Skipping cdn publish…"
+  else
+    # aws s3 cp build/bundle.js s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/$EXTENSION_NAME-$1.js --region us-west-1 --acl public-read
```
do we need this comment?

bernardwolff commented on tools/cdn.sh:
```diff
@@ -0,0 +1,42 @@
+    echo "There is already a $EXTENSION_NAME-$1.js in the cdn. Skipping cdn publish…"
+  else
+    # aws s3 cp build/bundle.js s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/$EXTENSION_NAME-$1.js --region us-west-1 --acl public-read
+    eccho "Running command (dry run)"
```
```suggestion
    echo "Running command (dry run)"
```

bernardwolff replied:
do we want a dry run or a real run?

joel-hamilton replied:
Since originally requesting review, I realized that the existing deployment tooling is definitely broken, so I'm adding that to this PR now. That's why the script has this debugging stuff. 

I'll put this PR in draft mode and re-request review once it's deploying as expected.

bernardwolff commented on tools/cdn.sh:
```diff
@@ -0,0 +1,42 @@
+  else
+    # aws s3 cp build/bundle.js s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/$EXTENSION_NAME-$1.js --region us-west-1 --acl public-read
+    eccho "Running command (dry run)"
+    echo "aws s3 cp build/bundle.js s3://assets.us.auth0.com/extensions/$EXTENSION_NAME/$EXTENSION_NAME-$1.js --region us-west-1 --acl public-read"
```
do we want to just run the `aws` command instead of echoing it?
